### ISSUE TITLE
buildbootstrapcli.sh needs version.cpp

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/source-build#331]
Executing buildbootstrapcli.sh fails in CMake of one of the projects due to a missing version.cpp file.  A simple `touch <path>/version.cpp` fixed the issue.

The directory path for the version.cpp file is outside of the `source-build` project.  I suspect the issue is that one of the projects is being built w/o `cd` to the expected directory.

@janvorli FYI

**Comment by janvorli** [comment]
@sdmaclea the issue is different. The version.cpp was added to core-setup repo recently (that's what has introduced the double version string presence that you've handled in your change to the bootstrap cli creation script - one is just the hash created by the old mechanism, the other is the full version info formatted the same way as in coreclr / corefx), but it is generated using msbuild. So in the bootstrap cli creation case, the file isn't created. 

In the coreclr, the build.sh has a fallback path that generates the file, it starts here:
https://github.com/dotnet/coreclr/blob/master/build.sh#L280

Corefx has a similar piece of code in the src/Native/build-native.sh:
https://github.com/dotnet/corefx/blob/master/src/Native/build-native.sh#L117

However, core-setup is missing that. So it seems we should add it to core-setup too in order to fix the issue. When I have tried to create a 2.1.x bootstrap cli some time ago, I've just created an empty version.cpp manually. And then I've forgotten about the problem.

cc: @eerhardt
